primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -812,4 +818,59 @@ public static Object convertToType(Object value, Class<?> valueType, Logger logg
             return value;
         }
     }
+
+    /**
+     * Invoke callback on the first rendered {@link EditableValueHolder} component
+     */
+    public static void invokeOnDeepestEditableValueHolder(FacesContext context, UIComponent composite, ContextCallback callback) {
```

Comment:
@tandraschko I think it could be renamed, turns out to be the closest as it takes the first rendered component WDYT?

Reply:
the closest means the "closest outer" one?
in this case we go into all nested composites and look for the deepest nested EditableValueHolder
therefore IMO its "deepest" and belongs only the composites, therefor it was in CompositeUtils.

Reply:
just to make sure:

if you do something like
```
p:outputLabel for=@next
myCC:wrappedWrappedInput
```

and wrappedWrappedInput has a cc:editableValueHolder for another CC called wrappedInput, which has a cc:editableValueHolder for a p:inputText

OutputLabelRenderer will then call the invokeOnDeepestEditableValueHolder and will be finally invoked for the real p:inputText

Reply:
> the closest means the "closest outer" one?

Deepest turns out to be fine, it's not accurate but it's fine. We actually don't look for the deepest, but the "first" rendered EditableValueHolder to be precise

> in this case we go into all nested composites and look for the deepest nested EditableValueHolder
therefore IMO its "deepest" and belongs only the composites, therefor it was in CompositeUtils.

It was indeed introduced because of composites, but again see #12519. CompositeUtils if it was only meant for composite, it's not the case, it's being used from simple component, to UIPanel component and for composite component.

Reply:
it was actually only meant only for composites; im also not happy with ComponentUtils but this belongs to the linked issue and this bugfix
i just dont like to mix refactorings and bugfixes

Reply:
I don't mind putting it back in CompositeUtils, and move it once https://github.com/primefaces/primefaces/issues/12519 gets solved

Reply:
That is a good idea @Rapster we can refactor with that other ticket.

Reply:
sounds good! +1